ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/LedgerClient.scala

```diff
@@ -77,38 +80,73 @@ object LedgerClient {
-
+  /**
+    * Constructs a [[Channel]], ensuring that the [[LedgerClientConfiguration]] is picked up and valid
+    *
+    * You'll generally want to use [[singleHost]], use this only if you need a higher level of control
+    * over your [[Channel]].
+    */
+  def constructChannel(
+      hostIp: String,
+      port: Int,
+      configuration: LedgerClientConfiguration): Try[Channel] = {
+    val builder: NettyChannelBuilder = NettyChannelBuilder.forAddress(hostIp, port)
     configuration.sslContext
       .fold {
-        builder.usePlaintext()
+        if (configuration.accessToken.isDefined) {
+          Failure(new IllegalArgumentException("Cannot send a token over an insecure channel"))
+        } else {
+          Success(builder.usePlaintext())
+        }
       } { sslContext =>
-        builder.sslContext(sslContext).negotiationType(NegotiationType.TLS)
+        Success(builder.sslContext(sslContext).negotiationType(NegotiationType.TLS))
       }
+      .map { builder =>
+        val channel = builder.build()
+        val _ = sys.addShutdownHook { val _ = channel.shutdownNow() }
+        channel
+      }
+  }
 
-    val channel = builder.build()
-
-    val _ = sys.addShutdownHook { val _ = channel.shutdownNow() }
+  /**
+    * A convenient shortcut to build a [[LedgerClient]]
+    */
+  def singleHost(hostIp: String, port: Int, configuration: LedgerClientConfiguration)(
+      implicit ec: ExecutionContext,
+      esf: ExecutionSequencerFactory): Future[LedgerClient] =
+    Future
+      .fromTry(constructChannel(hostIp, port, configuration))
+      .flatMap(forChannel(configuration, _))
+
+  val auth = Metadata.Key.of("Authorization", Metadata.ASCII_STRING_MARSHALLER)
+
+  def callCredentials(token: String): CallCredentials =
+    new CallCredentials {
+      override def applyRequestMetadata(
+          requestInfo: CallCredentials.RequestInfo,
+          appExecutor: Executor,
+          applier: CallCredentials.MetadataApplier): Unit = {
+        val metadata = new Metadata
+        metadata.put(auth, token)
+        applier.apply(metadata)
+      }
 
-    forChannel(configuration, channel)
+      // Should be a noop but never called; tries to make it clearer to implementors that they may break in the future.
+      override def thisUsesUnstableApi(): Unit = ()
+    }
 
-  }
   def forChannel(configuration: LedgerClientConfiguration, channel: Channel)(
       implicit ec: ExecutionContext,
       esf: ExecutionSequencerFactory): Future[LedgerClient] = {
+    val creds = configuration.accessToken.map(callCredentials).orNull
     apply(
-      LedgerIdentityServiceGrpc.stub(channel),
-      TransactionServiceGrpc.stub(channel),
-      ActiveContractsServiceGrpc.stub(channel),
-      CommandSubmissionServiceGrpc.stub(channel),
-      CommandCompletionServiceGrpc.stub(channel),
-      CommandServiceGrpc.stub(channel),
-      PackageServiceGrpc.stub(channel),
+      LedgerIdentityServiceGrpc.stub(channel).withCallCredentials(creds),
```

Comment:
I assume `.withCallCredentials(null)` is a no-op?

Reply:
It explicitly sets the `CallCredentials` to be `null`, so that they are not used.